executor/extension/validator/eth_state_test_log_hash_validator.go

```diff
@@ -47,11 +43,10 @@ type ethStateTestLogHashValidator struct {
 	cfg *utils.Config
 }
 
-func (e *ethStateTestLogHashValidator) PostTransaction(state executor.State[txcontext.TxContext], ctx *executor.Context) error {
+func (e *ethStateTestLogHashValidator) PostBlock(state executor.State[txcontext.TxContext], ctx *executor.Context) error {
```

Comment:
Logs should be emitted on a per-transaction level, not on a per-block level. I guess the switch is fine, since in the ethereum tests likely every transaction is processed in its own fake block, but logically this seems odd.

Why would this be switched to blocks?

Reply:
We need to call `EndBlock()` after each transaction because we need to call `stateDb.GetHash()`. So we found logical that we should treat is as `'one tx = one block'`

Reply:
We should still call this in `PostTransaction()` if `GetLogs` is a transaction level func...